### PR TITLE
Mention that there are other receiver types

### PR DESCRIPTION
**Pull request by mgeisler**
As per the discussion in #70.

### CODE REVIEW

File: src/methods/receiver.md
Comment (maurer) [inline src/methods/receiver.md:17]: Consider referencing https://doc.rust-lang.org/reference/special-types-and-traits.html so the class can see the enumeration of special receiver types?  (Rc, Arc, Pin are the remaining ones for the moment).

Comment (maurer) [inline src/methods/receiver.md:17]: Probably rephrase this to make it clear where these types come from, and that they're receiver types, e.g.

"Beyond variants on `self`, there are also [special](https://doc.rust-lang.org/reference/special-types-and-traits.html) wrapper types allowed to be receiver types, such as `Box<Self>`.

Comment (maurer) [inline src/methods/receiver.md:17]: Should we have a discussion of why you'd want to use alternate receiver types rather than relying on `impl Deref`, or do we want to rely on the instructor for that?

**Review by maurer — APPROVED** [review]
LGTM as long as you do something with the sentence to make it clearer that it's part of the receiver discussion - it's clear in the code because it's part or `receiver.md`, but it's not part of the bulleted list on receivers, the sentence doesn't mention receivers, etc.

I've also made some other comments that would improve it, but up to you on those.

File: src/methods/receiver.md
Comment (mgeisler) [reply src/methods/receiver.md]: Great idea to include a link to that page.

Comment (mgeisler) [reply src/methods/receiver.md]: I think this is something we'll have to defer to the speaker notes (i.e., to the instructor as you suggest).

As you hint, the pages themselves cannot go into detail on all topics — plus, people are seeing this on Day 2 after having been exposed to Rust for 5-6 hours. I think we should stick to the main point: how does a method look in Rust and related and when do you need to use `&mut self` instead of just `&self`.

Comment (mgeisler) [reply src/methods/receiver.md]: Thanks, that's a nice way to put it, I updated the PR to say that.

**Pull request merged**
